pyproject.toml

```diff
@@ -256,6 +258,12 @@ ignore = [
     "TID252",  # Relative imports from parent modules are banned
     # https://beta.ruff.rs/docs/rules/#flake8-use-pathlib-pth
     "PTH123", # pathlib-open - this would force pathlib usage anytime open or with open was used.
+    # https://beta.ruff.rs/docs/rules/#flake8-bugbear-b
+    # TODO: enable these
+    "B904", # raise-without-from-inside-except
+    "B028", # no-explicit-stacklevel - https://beta.ruff.rs/docs/rules/no-explicit-stacklevel/
+    "B007", # unused-loop-control-variable
+    "B009", # get-attr-with-constant
```

Comment:
These rules have a significant number of violations.
Too many to properly review in this PR.
```
$ ruff . --statistics
249	B904	[ ] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
 82	B028	[ ] No explicit `stacklevel` keyword argument found
 38	B007	[*] Loop control variable `attr` not used within loop body
 15	B009	[*] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.
```